src/settings.rs

```diff
@@ -35,35 +63,44 @@ pub struct Network {
     pub ws_uri: String,
 }
 
+
 impl Settings {
     /// Construct a new settings object from a file path and the environment
     ///
     /// # Arguments
     ///
     /// * `path` - Path to a configuration file
-    pub fn new<P>(path: Option<P>) -> Result<Self>
+    pub fn new<P>(path: Option<P>) -> Result<Self, Error>
     where
         P: AsRef<Path>,
     {
         let mut c = Config::new();
 
+        // We won't need to derive implementation from enum discriminator so
+        // don't bother implementing conversion for ValueKinds.
+        c.set_default("logging.format", LogFmt::JSON.as_str())?;
+
         c.set_default("relay.confirmations", 12)?;
         c.set_default("relay.anchor_frequency", 100)?;
 
         if let Some(p) = path {
-            let ps = p.as_ref().to_str().chain_err(|| ErrorKind::InvalidConfigFilePath)?;
-            c.merge(File::with_name(ps))?;
+            let ps = p.as_ref().to_str();
+            if let Some(path_str) = ps {
+                c.merge(File::with_name(path_str))?;
+            } else {
+                return Err(ConfigError::InvalidConfigFilePath.into());
+            }
         }
 
         c.merge(Environment::new())?;
         c.try_into().map_err(|e| e.into()).and_then(|s: Self| s.validated())
     }
 
-    fn validated(self) -> Result<Self> {
+    fn validated(self) -> Result<Self, Error> {
```

Comment:
Think it would be cleaner if this returned Result<Self, ConfigError> and the above closure map_err'ed its result as well.

```
         c.try_into()                                            
             .map_err(|e| e.into())  
             .and_then(|s: Self| s.validated().map_err(|e| e.into())) 
```